src/libtiled/mapreader.cpp

```diff
@@ -897,23 +952,30 @@ void MapReaderPrivate::decodeCSVLayerData(TileLayer &tileLayer,
     QString trimText = text.trimmed().toString();
     QStringList tiles = trimText.split(QLatin1Char(','));
 
-    if (tiles.length() != tileLayer.width() * tileLayer.height()) {
+    int lengthCheck = (mMap->infinite()) ? CHUNK_SIZE * CHUNK_SIZE :
```

Comment:
`decodeCSVLayerData` should take a `QRect` parameter specifying the size and position of the tiles it is decoding.